tagstudio/src/core/library.py

```diff
@@ -276,18 +279,19 @@ def __repr__(self) -> str:
         return self.__str__()
 
     def __eq__(self, __value: object) -> bool:
+        __value = cast(Self, __value)
         if os.name == "nt":
             return (
-                int(self.id) == int(__value.id_)
-                and self.filename.lower() == __value.filename.lower()
-                and self.path.lower() == __value.path.lower()
+                int(self.id) == int(__value.id)
+                and self.filename.lower() == __value.filename.lower()  # type: ignore
+                and self.path.lower() == __value.path.lower()  # type: ignore
```

Comment:
looking at this, I'm not convinced this has ever worked, because this object looks like it doesnt have `.filename` nor `.path` property 🤔